### PR TITLE
Development: Add VS Code Devcontainer Configuration

### PR DESCRIPTION
**Pull request by p-h-a-i-l**
## Proposed change

This pull request adds configuration files for Visual Studio Code devcontainers. Devcontainers provide a consistent and isolated development environment, ensuring that all necessary dependencies and tools are readily available. This setup simplifies the onboarding process for new contributors by reducing the complexity of configuring their development environment, thereby encouraging more contributions.

Closes #6888

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [ ] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [x] Other. Please explain: Lowers the barrier for contribution by providing a pre-configured development environment. This change has no impact on production code as it only includes configuration files for development setup.

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by stumpylog** [comment]
Generally seems like a good idea.  Can some of the files be integrated into the existing workspace file?

**Comment by p-h-a-i-l** [comment]
> Generally seems like a good idea. Can some of the files be integrated into the existing workspace file?

The vscode config files in .devcontainer/vscode are not in use on the bare metal route. docker-compose.devcontainer.sqlite-tika.yml just mounts it inside the devcontainer as the active vscode configuration. 
The idea was to avoid interference with bare metal users, but provide easy onboarding for devcontainer users by providing tasks and launch configurations.

**Review by stumpylog — CHANGES_REQUESTED** [review]

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        --output jbig2enc_${JBIG2ENC_VERSION}-1_${TARGETARCH}.deb \
+        https://github.com/paperless-ngx/builder/releases/download/jbig2enc-${JBIG2ENC_VERSION}/jbig2enc_${JBIG2ENC_VERSION}-1_${TARGETARCH}.deb \
+      && dpkg --install ./jbig2enc_${JBIG2ENC_VERSION}-1_${TARGETARCH}.deb
+
+# setup docker-specific things
+# These change sometimes, but rarely
+WORKDIR /usr/src/paperless/src/docker/
+
+COPY [ \
+  "docker/imagemagick-policy.xml", \
```
[the remark this answers is not in the document]
Comment (p-h-a-i-l) [reply .devcontainer/Dockerfile:126]: done

File: .devcontainer/docker-compose.devcontainer.sqlite-tika.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# docker-compose file for running paperless from the docker container registry.
```
[the remark this answers is not in the document]
Comment (p-h-a-i-l) [reply .devcontainer/docker-compose.devcontainer.sqlite-tika.yml]: done

File: .devcontainer/docker-compose.devcontainer.sqlite-tika.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+# - Copy this file as 'docker-compose.yml' and the files 'docker-compose.env'
+#   and '.env' into a folder.
+# - Run 'docker-compose pull'.
+# - Run 'docker-compose run --rm webserver createsuperuser' to create a user.
+# - Run 'docker-compose up -d'.
+#
+# For more extensive installation and update instructions, refer to the
+# documentation.
+
+version: "3.4"
```
[the remark this answers is not in the document]
Comment (p-h-a-i-l) [reply .devcontainer/docker-compose.devcontainer.sqlite-tika.yml]: removed

File: .devcontainer/docker-compose.devcontainer.sqlite-tika.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      PAPERLESS_TIKA_GOTENBERG_ENDPOINT: http://gotenberg:3000
+      PAPERLESS_TIKA_ENDPOINT: http://tika:9998
+      PAPERLESS_STATICDIR: ./src/documents/static
+      PAPERLESS_DEBUG: true
+
+    # Overrides default command so things don't shut down after the process ends.
+    command: /bin/sh -c "chown -R paperless:paperless /usr/src/paperless/paperless-ngx/src/documents/static/frontend && chown -R paperless:paperless /usr/src/paperless/paperless-ngx/.ruff_cache &&while sleep 1000; do :; done"
+
+  gotenberg:
+    image: docker.io/gotenberg/gotenberg:7.8
```
[the remark this answers is not in the document]
Comment (p-h-a-i-l) [reply .devcontainer/docker-compose.devcontainer.sqlite-tika.yml]: done

**Comment by lobochrome** [comment]
I would love this. I always wanted to commit some japanese specific parsing code - but never could get over installing the dev environment.

**Comment by p-h-a-i-l** [comment]
> I would love this. I always wanted to commit some japanese specific parsing code - but never could get over installing the dev environment.

Glad you Like it! If you want to help, I am looking for some testers 😬
Maybe there are improvements to be eine in the documentation..

**Comment by lobochrome** [comment]
I can't commit to proper "testing". But I can sure give it a spin if I find an evening ;).

**Review by stumpylog — APPROVED** [review]
Tested it out and everything seemed to be working alright.  If you could just update the branch, I'll merge this in

**Pull request merged**